lib/puppet/functions/stdlib/to_json_pretty.rb

```diff
@@ -67,8 +67,10 @@ def to_json_pretty(data, skip_undef = false, opts = nil)
     end
 
     data = data.compact if skip_undef && (data.is_a?(Array) || Hash)
-    # Call ::JSON to ensure it references the JSON library from Ruby's standard library
-    # instead of a random JSON namespace that might be in scope due to user code.
-    JSON.pretty_generate(data, opts) << "\n"
+    call_function('stdlib::rewrap_sensitive_data', data) do |unwrapped_data|
+      # Call ::JSON to ensure it references the JSON library from Ruby's standard library
+      # instead of a random JSON namespace that might be in scope due to user code.
+      ::JSON.pretty_generate(unwrapped_data, opts) << "\n"
```

Comment:
Leading colons put back in to match existing comment.  See https://github.com/puppetlabs/puppetlabs-stdlib/pull/1307#discussion_r1489325529